util/misc.go

```diff
@@ -53,3 +55,20 @@ func GetStack() []byte {
 	buf = buf[:stackSize]
 	return buf
 }
+
+// WithRecovery wraps goroutine startup call with recovery handle.
+// There are two out of box common handlers `WithPanicLogf` and `WithPanicMetricInc`,
+// Caller can custom define new handler use last optional parameters.
+func WithRecovery(ctx context.Context, fn func(ctx context.Context), handler func(r interface{})) {
```

Comment:
I think the `ctx` parameter can be removed and this function can be simplified to:
```go
func WithRecovery(callee func(), handler func(r interface{})) {
}
```